src/main/java/org/folio/rest/impl/BaseApi.java

```diff
@@ -85,27 +85,37 @@ protected HttpException handleProcessingError(Throwable throwable) {
   }
 
   public Response buildErrorResponse(Throwable throwable) {
-    return buildErrorResponse(handleProcessingError(throwable));
+  logger.error("Exception encountered", throwable.getCause());
+    final int code = defineErrorCode(throwable);
+    final Errors errors = convertToErrors(throwable);
+    final Response.ResponseBuilder responseBuilder = createResponseBuilder(code);
+    return responseBuilder.header(CONTENT_TYPE, APPLICATION_JSON)
+      .entity(errors)
+      .build();
   }
 
-  public Response buildErrorResponse(HttpException exception) {
-    final Response.ResponseBuilder responseBuilder;
-    switch (exception.getCode()) {
+  public static int defineErrorCode(Throwable throwable) {
```

Comment:
nitpick: this method could return `javax.ws.rs.core.Response.Status` rather than its status code. The switch statement in `createResponseBuilder` method below would operate on `javax.ws.rs.core.Response.Status` rather than `int`. To clarify, this method would return `INTERNAL_SERVER_ERROR` object instead of its error code.

The code smell became more obvious because the the default return in the switch statement still relies on `javax.ws.rs.core.Response.Status`.

Reply:
i have updated the logic for error  response building and removed this method , requesting you to please review